core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -197,7 +193,8 @@
   private volatile boolean terminated;
   private final CountDownLatch terminatedLatch = new CountDownLatch(1);
 
-  private final ManagedChannelReference phantom;
+  // If set, must be set immediately after instantiation
+  private Reference<?> phantom;
```

Comment:
I think this should be volatile, since creation and shutdown may be on different threads

Reply:
It is safe as-is. Although verifying whether it is set immediately after instantiation requires looking at multiple classes.